src/jsp/antlr4/PcsParser.g4

```diff
@@ -0,0 +1,615 @@
+/**
+ * CUBRID PL/CSQL Parser grammar based on and updated from
+ *  Oracle(c) PL/SQL 11g Parser (https://github.com/antlr/grammars-v4/tree/master/sql/plsql)
+ *
+ * Copyright (c) 2009-2011 Alexandre Porcelli <alexandre.porcelli@gmail.com>
+ * Copyright (c) 2015-2019 Ivan Kochurkin (KvanTTT, kvanttt@gmail.com, Positive Technologies).
+ * Copyright (c) 2017 Mark Adams <madams51703@gmail.com>
+ * Copyright (c) 2016 CUBRID Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+parser grammar PcsParser;
+import SqlParser;
+
+@header {
+package com.cubrid.plcsql.compiler.antlrgen;
+}
+
+options {
+    tokenVocab=PcsLexer;
+}
+
+sql_script
+    : unit_statement EOF
+    ;
+
+unit_statement
+    : create_function
+    | create_procedure
+    ;
+
+create_function
+    : CREATE (OR REPLACE)? FUNCTION identifier ('(' parameter_list ')')? RETURN type_spec
+      (IS | AS) seq_of_declare_specs? body ';'
+    ;
+
+create_procedure
+    : CREATE (OR REPLACE)? PROCEDURE identifier ('(' parameter_list ')')?
+      (IS | AS) seq_of_declare_specs? body ';'
+    ;
+
+function_body
+    : FUNCTION identifier ('(' parameter_list ')')? RETURN type_spec
+      (IS | AS) seq_of_declare_specs? body ';'
+    ;
+
+procedure_body
+    : PROCEDURE identifier ('(' parameter_list ')')?
+      (IS | AS) seq_of_declare_specs? body ';'
+    ;
```

Comment:
It is not compatible with Oracle syntax. though It looks consistent with the current CUBRID syntax.
It can cause inconvenience in Oracle PL/SQL code migration.
We need to talk about this and make a decision.

Reply:
Migration tools should consider compatibility with Oracle PL/SQL, and IMO,  we don't have to modify our stored routine syntax for that.

Reply:
As far as I remember, we have not discussed and decided the syntax with 'language plcsql'. Have we?

Reply:
Without a syntax like 'langauge plcsql', how does the parser differentiate between javasp and plcsql in stored routine definitions? I remember the discussion at https://github.com/CUBRID/cubrid-manual/commit/180f074e2f89cd1d9a2710affae6562e4e35a200 was about the functional requirements of plcsql we would support. The design discussed with @mhoh3963 was to send only the procedure/function's body part between BEGIN ... END to the plcsql compiler, not the entire CREATE ... statement. For compatibility with procedure/function definition syntax, it would be good to clarify this.

Reply:
We should differenciate those two cases without too much effort because create procedure/function for javasp does not contain begin ... end part.
The compiler must know the procedure/function's name, formal parameters, and declaration parts before the begin ... end part. The easiest way is for the compiler to receive not only the body part but the whole create procedure/function statement. I guess mhoh's intention was indeed 'unparsed string' rather than 'the body only'. Of course, we need to clarify this.